src/btree.c

```diff
@@ -6551,6 +6556,23 @@ static int allocateBtreePage(
   return rc;
 }
 
+static int allocateBtreePage(
+  BtShared *pBt,         /* The btree */
+  MemPage **ppPage,      /* Store pointer to the allocated page here */
+  Pgno *pPgno,           /* Store the page number here */
+  Pgno nearby,           /* Search for a page near this one */
+  u8 eMode               /* BTALLOC_EXACT, BTALLOC_LT, or BTALLOC_ANY */
+){
+  if (pBt->pPager->pVfs->xAllocatePage) {
+    int rc = pBt->pPager->pVfs->xAllocatePage(pBt->pPager->fd, pPgno, eMode);
+    if (!rc) {
+      return rc;
+    }
+    return btreeGetUnusedPage(pBt, *pPgno, ppPage, 0);
```

Comment:
It would require the implementors to use structures that reside in *Int.h headers, and are not public, e.g. MemPage or BtShared. I'd strongly prefer to avoid exposing any internal details and keep the interface as clean as possible. Also, our first target use case, storing the free page info in FoundationDB, is a prime example why we shouldn't need any of this private info.

Reply:
Aah, right. The asymmetry is unfortunate, but makes perfect sense.